lib/modules/manager/maven/update.ts

```diff
@@ -69,6 +69,20 @@ export function updateAtPosition(
     // TODO: validate newValue (#22198)
     const replacedPart = versionPart.replace(version, newValue!);
     return leftPart + replacedPart + restPart;
+  } else {
```

Comment:
Add code comment documenting when/why this logic branch would occur

Reply:
I guess that is explaining enough ;-).